.github/workflows/ci-cd.yml

```diff
@@ -41,20 +41,17 @@ jobs:
         os:
         - ubuntu-24.04
         - ubuntu-24.04-arm
-        include:
-        - experimental: false
-        - upload-coverage: false
-        - upload-coverage: true
-          python-version: 3.11
-          os: ubuntu-24.04
+        experimental:
+        - false
       fail-fast: false
     uses: ./.github/workflows/reusable-test.yml
     with:
       python-version: ${{ matrix.python-version }}
       os: ${{ matrix.os }}
       continue-on-error: ${{ matrix.experimental }}
       enable-cache: ${{ github.ref_type == 'tag' && 'false' || 'auto' }}
-      upload-coverage: ${{ matrix.upload-coverage }}
+      upload-coverage:
+        ${{ matrix.python-version == 3.11 && matrix.os == 'ubuntu-24.04' }}
     secrets:
       # yamllint disable-line rule:line-length
       codecov-token: ${{ matrix.upload-coverage && secrets.CODECOV_TOKEN || '' }}
```

Comment:
Sorry, but we need to change that line also, if we want to get rid of `upload-coverage`.

Reply:
We could fast-track #1367 to resolve this issue.

Reply:
Why can't we just merge this one first?

Reply:
Let's patch it instead, then.